### DISCUSSION TITLE
Sort by date rather than order entered?

### DISCUSSION THREAD

**Discussion by jameswestman** [NickvisionApps/Denaro#196] [Ideas]
There's an option to sort ascending or descending, but it doesn't look like there's a way to sort by date rather than the order the transactions are entered. I'd like to add a bunch of older transactions from before I started using the app, but I'd prefer they don't show up before all my other transactions that really are recent.

Thanks for this wonderful app btw!

**Comment by nlogozzo** [comment]
@fsobolev Maybe we keep the same `Sort First to Last` and `Sort Last to First` toggles but in the controller order by date and not id?

**Reply by nlogozzo** [reply]
Or maybe add an option to `AccountSettingsDialog` that says `Sort Transactions By`: `Id` or `Date` and then use it that way? 

Probably this way is the best option.

**Reply by fsobolev** [reply]
We can add Gtk.DropDown next to sorting toggles that will contain "Sort by Id" and "Sort by date" options that will be remembered in account metadata just like sorting order.

**Reply by fsobolev** [reply]
@nlogozzo 

![](https://i.imgur.com/rW1nBG5.png)

**Reply by nlogozzo** [reply]
That looks great! I will implement the backend/controller for sort by and then you can wire up the GNOME and commit it.

I'll make a `sort-by` branch.

**Comment by nlogozzo** [comment]
Use #197 to track.
